core/controllers/admin.py

```diff
@@ -588,6 +589,43 @@ def _generate_dummy_explorations(
             raise Exception('Cannot generate dummy explorations in production.')
 
 
+class AdminMathSvgImageGenerationHandler(base.BaseHandler):
+    """Handler updating Explorations having math rich-text components with
+    math SVGs.
+    """
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        latex_strings_to_exp_id_mapping = (
+            exp_services.get_latex_strings_and_exp_ids_for_generating_svgs())
+        self.render_json({
+            'result': latex_strings_to_exp_id_mapping
+        })
+
+    @acl_decorators.can_access_admin_page
+    def post(self):
+        latex_to_svg_mappings = self.payload.get('latexMapping')
+        for exp_id, latex_to_svg_mapping_dict in latex_to_svg_mappings.items():
+            for latex_string in latex_to_svg_mapping_dict.keys():
+                image = self.request.get(
+                    latex_to_svg_mappings[exp_id][latex_string]['latexId'])
+                if not image:
+                    raise self.InvalidInputException(
+                        'In Exploration %s SVG for latex value %s not supplied.'
+                        % (exp_id, latex_string))
+                latex_to_svg_mappings[exp_id][latex_string]['file'] = image
+
+        no_of_explorations_updated = 0
+        for exp_id in latex_to_svg_mappings.keys():
+            exp_services.update_explorations_with_math_svgs(
```

Comment:
Is it worth trying to do this using a single function for all the exp IDs, so that you can do put_multi, get_multi, etc.? I'm not sure if the complexity is worth it.

Approximately how many explorations will you be handling at a time? 100 individual puts in sequence is not great.

Try making some estimates and figure out what to do here based on that.

Reply:
I'm actually not sure if multi_put is used anywhere (or defined) in the codebase for versioned models.
But from audit job results, there are 340 explorations needed to be migrated with 4895 SVG images required.

> 100 individual puts in sequence is not great

So I don't think this would be needed.
Even if we plan to do it in 10 batches we need to update around 30 explorations at a time